src/client/datascience/commands/commandRegistry.ts

```diff
@@ -496,11 +498,12 @@ export class CommandRegistry implements IDisposable {
     }
     private async onVariablePanelShowDataViewerRequest(request: IShowDataViewerFromVariablePanel) {
         sendTelemetryEvent(EventName.OPEN_DATAVIEWER_FROM_VARIABLE_WINDOW_REQUEST);
-        if (this.debugService.activeDebugSession) {
-            const jupyterVariable = convertDebugProtocolVariableToIJupyterVariable(
-                request.variable as DebugProtocol.Variable
-            );
+        if (this.debugService.activeDebugSession && this.variableProvider.getFullVariable) {
```

Comment:
Doesn't this mean, that every one implementing the variable provider MUST also implement `getFullVariable`. Isn't this a breaking change?
Is this ok for AML & others we consume our API?

Reply:
I added it as an optional function in the interface definition for `IDataViewerDataProvider`, and this function is only used for debugger integration, so it should not be a problem for AML.

Reply:
If that's the case, then why do we need to make it optional?

Reply:
Just for additional type safety to communicate that that function is not standard and not guaranteed to be implemented by all IDataViewerDataProviders. I should probably also add a comment to that effect. Otherwise future users of this interface may not realize that this function should not be used in showDataViewer codepaths.

Reply:
Wow, sorry, I was thinking of the wrong interface. `getFullVariable` is implemented on `IJupyterVariable`, not `IDataViewerDataProvider`. You're totally right, this function doesn't need to be optional at all because AML will never see it (it's specific to JupyterVariableDataProvider which is our implementation of IDataViewerDataProvider).